phf_shared/src/lib.rs

```diff
@@ -364,62 +364,44 @@ impl PhfHash for char {
 }
 
 // minimize duplicated code since formatting drags in quite a bit
-fn fmt_array(array: &[u8], f: &mut fmt::Formatter<'_>) -> fmt::Result {
+fn fmt_array<T: std::fmt::Debug>(array: &[T], f: &mut fmt::Formatter<'_>) -> fmt::Result {
     write!(f, "{:?}", array)
 }
 
 macro_rules! array_impl (
-    ($t:ty, $n:expr) => (
-        impl PhfHash for [$t; $n] {
+    ($t:ty) => (
+        impl<const N: usize> PhfHash for [$t; N] {
             #[inline]
             fn phf_hash<H: Hasher>(&self, state: &mut H) {
-                state.write(self);
+                for v in self {
+                    v.phf_hash(state);
+                }
             }
         }
 
-        impl FmtConst for [$t; $n] {
+        impl<const N: usize>  FmtConst for [$t; N] {
```

Comment:
```suggestion
        impl<const N: usize> FmtConst for [$t; N] {
```

The formatting is a bit broken, could you run rustfmt?

Reply:
Sure, just fixed.